### ISSUE TITLE
[Bug] Use dataset for multiple series pie chart, render wrong

### ISSUE THREAD

**Issue by zhoudaxia2016** [apache/echarts#19262]
### Version

5.4.3

### Link to Minimal Reproduction

https://echarts.apache.org/examples/en/editor.html?c=bar-polar-label-tangential&code=PYBwLglsB2AEC8sDeAoWsAmBDMWDOApmAFzJrqx7ACuATgMYGkDa5FszA5LJwDQ9Y-PAEacAurzYUuANywAbagQCMQ5fwDMEqelkKlAJiEH-BsVPPoAvpPSFaEAnhZTU7dNCwBbJgL47YMABPEF9OEEd_d1haLAwIamcOTgBWAFIhTg108VsKG1cAzx9STnoo92DQ0oiCCvZY-MSWTgA2AAYM_k5lds7cqStycysAbiA

### Steps to Reproduce

Open the Minimal Reproduction link.

### Current Behavior

https://echarts.apache.org/examples/en/editor.html?c=bar-polar-label-tangential&code=PYBwLglsB2AEC8sDeAoWsAmBDMWDOApmAFzJrqx7ACuATgMYGkDa5FszA5LJwDQ9Y-PAEacAurzYUuANywAbagQCMQ5fwDMEqelkKlAJiEH-BsVPPoAvpPSFaEAnhZTU7dNCwBbJgL47YMABPEF9OEEd_d1haLAwIamcOTgBWAFIhTg108VsKG1cAzx9STnoo92DQ0oiCCvZY-MSWTgA2AAYM_k5lds7cqStycysAbiA
![bar-polar-label-tangential (1)](https://github.com/apache/echarts/assets/24446852/d4782ca3-5477-47ad-a2b1-404959ac3216)

### Expected Behavior

https://echarts.apache.org/examples/en/editor.html?c=bar-polar-label-tangential&code=PYBwLglsB2AEC8sDeAoWsDOBTAThLGAXLANprrLkWzQCGAtlsQOS3MA0VFYAniE7GYh8HLugAmtMLWIkkAN1oAbAK4CAjOxoMBzRaqzrmAXy0Lla4gCYtdRi31qrJgLqdq6HLXEQVRUswArACkHIIAzCHMblymXKge2vaCAEaiibz8LMJY6R6S0rLmBsThtjoOFoYmZo4CNkm6dc7GMYlePn6yzABsAAyhWszqfQPR7hTG5C4oxgDcQA
![bar-polar-label-tangential](https://github.com/apache/echarts/assets/24446852/5d12b53c-9b7c-48bb-bb80-d125266b32d1)

### Environment

```markdown
- OS:
- Browser:
- Framework:
```

### Any additional comments?

Do I generate a wrong `dataset.source`?
I know I can use `series.data` to achieve this，but I want all my charts use the same dataset option to generate data.
I also find out that I can use dataset with `series.encode`, but I want to figure it out whether it's a bug or not.

**Comment by helgasoft** [comment]
>  I want all my charts use the same dataset option to generate data.

possible with a multi-dataset - [demo](https://echarts.apache.org/examples/en/editor.html?c=bar-polar-label-tangential&code=PYBwLglsB2AEC8sDeAoWsAmBDMWDOApmAFywDaa6ysewArgE4DGBpZSAblgDZ2uwBGADSxoWALb8A5F14EBUgL5DOPPqQBMIsZNIy1BDUoC6yyuiQ16zfu1nrYAZm0Tp9-UpXvNL3bH1yRoqmQpTGoeiEDBAEeGzmyAnoOtJYUhFU6GAAniDSIDHpSbAMWBgQdHHkUgCsAKTp_o71UuEJZpmomcmuekyN2LiEYACS0BgEAB7EwsU5eXoFBI3FpeWVbFIAbAAMDSJSAJx7rRnoimEoigDcQA)
NB: please close issue if problem solved.

**Comment by zhoudaxia2016** [comment]
> > I want all my charts use the same dataset option to generate data.
> 
> possible with a multi-dataset - [demo](https://echarts.apache.org/examples/en/editor.html?code=PYBwLglsB2AEC8sDeAoWsAmBDMWDOApmAFywDaa6ysewArgE4DGBpZZsAjADSwDkANywAbOgU58Autw4AmXoJFjZUyQF9uldEhr1mrchwDMCoaPFSZsefzPLVGytMqEGEAnjZbk39NCwAtgZ8WHyaVFRgAJ4gwSDuYb6wDFgYEHSe5HwArACkYfxGeZbejhGoEX6BwUwF2LiEYACS0BgEAB7EPEnRsaR88QQFSSlpGWx8AGwADPkKAJyzJRFqTihqANxAA) NB: please close issue if problem solved.

I want to avoid changing the dataset and wonder why one dataset can not achieve this.

**Comment by helgasoft** [comment]
> I want to avoid changing the dataset and wonder why one dataset can not achieve this.

it can, but you need to use _encode_ - [demo](https://echarts.apache.org/examples/en/editor.html?c=bar-polar-label-tangential&code=PYBwLglsB2AEC8sDeAoWsAmBDMWDOApmAFzJrqx7ACuATgMYGkDa5FszA5NJwDSycsfAQCNOAXV5sKXAG5YANtQIBGYSv4BmSdPRzFygEzDD_Q-OkX0AXynpCtCATwtpqdumhYAtkwFC7DzAATxA_ThAnYQJoemAMPyQIMAJvADkfcJ5-eSUmQU5bXVhaLAwIahcOTgBWAFJhTk16iUCbNrIPWC9fUk56PmKQsL7IgmjY-MTk1IzegWzYXOViTjEirtLyypZOADYABgb-TgBOI9bpa3IdawBuIA)

> I also find out that I can use dataset with series.encode, but I want to figure it out whether it's a bug or not.

not-a-bug, series [default layout](https://echarts.apache.org/en/option.html#series-pie.seriesLayoutBy) is by 'column', so you need to use _encode_ to assign different columns to multiple series.

**Comment by zhoudaxia2016** [comment]
> > I want to avoid changing the dataset and wonder why one dataset can not achieve this.
> 
> it can, but you need to use _encode_ - [demo](https://echarts.apache.org/examples/en/editor.html?c=bar-polar-label-tangential&code=PYBwLglsB2AEC8sDeAoWsAmBDMWDOApmAFzJrqx7ACuATgMYGkDa5FszA5NJwDSycsfAQCNOAXV5sKXAG5YANtQIBGYSv4BmSdPRzFygEzDD_Q-OkX0AXynpCtCATwtpqdumhYAtkwFC7DzAATxA_ThAnYQJoemAMPyQIMAJvADkfcJ5-eSUmQU5bXVhaLAwIahcOTgBWAFJhTk16iUCbNrIPWC9fUk56PmKQsL7IgmjY-MTk1IzegWzYXOViTjEirtLyypZOADYABgb-TgBOI9bpa3IdawBuIA)
> 
> > I also find out that I can use dataset with series.encode, but I want to figure it out whether it's a bug or not.
> 
> not-a-bug, series [default layout](https://echarts.apache.org/en/option.html#series-pie.seriesLayoutBy) is by 'column', so you need to use _encode_ to assign different columns to multiple series.

All the other charts's series default layout is by 'column' too. Why pie chart is different?

> > I want to avoid changing the dataset and wonder why one dataset can not achieve this.
> 
> it can, but you need to use _encode_ - [demo](https://echarts.apache.org/examples/en/editor.html?c=bar-polar-label-tangential&code=PYBwLglsB2AEC8sDeAoWsAmBDMWDOApmAFzJrqx7ACuATgMYGkDa5FszA5NJwDSycsfAQCNOAXV5sKXAG5YANtQIBGYSv4BmSdPRzFygEzDD_Q-OkX0AXynpCtCATwtpqdumhYAtkwFC7DzAATxA_ThAnYQJoemAMPyQIMAJvADkfcJ5-eSUmQU5bXVhaLAwIahcOTgBWAFJhTk16iUCbNrIPWC9fUk56PmKQsL7IgmjY-MTk1IzegWzYXOViTjEirtLyypZOADYABgb-TgBOI9bpa3IdawBuIA)
> 
> > I also find out that I can use dataset with series.encode, but I want to figure it out whether it's a bug or not.
> 
> not-a-bug, series [default layout](https://echarts.apache.org/en/option.html#series-pie.seriesLayoutBy) is by 'column', so you need to use _encode_ to assign different columns to multiple series.

Other charts's series default layout is by 'column' too. Why just pie chart needs to use encode.

**Comment by helgasoft** [comment]
Agree, pie chart is different. Maybe the others (line,bar..) have been "optimized" and pie hasn't.
But still, a missing feature is not a bug.
